source/client/stream_decoder.cc

```diff
@@ -118,20 +118,26 @@ void StreamDecoder::onPoolReady(Envoy::Http::RequestEncoder& encoder,
   stream_info_.upstreamInfo()->upstreamTiming().onFirstUpstreamTxByteSent(
       time_source_); // XXX(oschaaf): is this correct?
   const Envoy::Http::Status status =
-      encoder.encodeHeaders(*request_headers_, request_body_size_ == 0);
+      encoder.encodeHeaders(*request_headers_, false);
```

Comment:
Why do you hard coded request_body_size==0 to false  here? This changes the current behavior and assumes that every request has the request_body_size != 0. Is that by intention?